app/src/docs/v1.api-spec.yaml

```diff
@@ -1135,6 +1502,54 @@ components:
             permCode:
               $ref: '#/components/schemas/PermCode'
         - $ref: '#/components/schemas/DB-TimestampUserData'
+    DB-BucketPermission:
+      allOf:
+        - type: object
+          required:
+            - id
+            - bucketId
+            - userId
+            - permCode
+            - createdBy
+          properties:
+            id:
+              type: string
+              format: uuid
+              description: The unique identifier for this permission tuple
+              example: ac246e31-c807-496c-bc93-cd8bc2f1b2b4
+            objectId:
+              type: string
+              format: uuid
+              description: The unique identifier for the object
+              example: ac246e31-c807-496c-bc93-cd8bc2f1b2b4
+            userId:
+              type: string
+              format: uuid
+              description: The unique identifier of the user
+              example: ac246e31-c807-496c-bc93-cd8bc2f1b2b4
+            permCode:
+              $ref: '#/components/schemas/PermCode'
+            createdBy:
+              type: string
+              format: uuid
+              description: The unique identifier of the user
+              example: ac246e31-c807-496c-bc93-cd8bc2f1b2b4
+            createdAt:
+              type: string
+              format: yyyy-mm-ddThh:mm:ssZ
+              description: The created by date
+              example: 2022-12-02T18:13:09.954Z
+            updatedBy:
+              type: string
+              format: uuid
+              description: The unique identifier of the user
+              example: ac246e31-c807-496c-bc93-cd8bc2f1b2b4
+            updatedAt:
+              type: string
+              format: yyyy-mm-ddThh:mm:ssZ
+              description: The created by date
+              example: 2022-12-02T18:13:09.954Z
+        - $ref: '#/components/schemas/DB-TimestampUserData'
```

Comment:
These 4 attributes do not need to be re-declared as you're already pulling in `#/components/schemas/DB-TimestampUserData` on Line 1552.